spidermon/contrib/validation/cerberus/validator.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import absolute_import
+from cerberus.validator import (
+    Validator as cerberus_validator,
+    DocumentError,
+    SchemaError,
+)
+from spidermon.contrib.validation.validator import Validator
+from .translator import CerberusMessageTranslator
+
+
+class CerberusValidator(Validator):
+    default_translator = CerberusMessageTranslator()
+    name = "Cerberus"
+
+    def __init__(self, schema, translator=None, use_default_translator=True):
+        super(CerberusValidator, self).__init__(
+            translator=translator, use_default_translator=use_default_translator
+        )
+        self._schema = schema
+
+    def _validate(self, data, strict=False):
+        try:
+            validator = cerberus_validator(self._schema)
+            validator.validate(data)
+            for field_name, error in validator.errors.items():
+                self._add_errors({field_name: error})
+        except DocumentError:
+            raise ValueError("{} is not a document, must be of dict type".format(data))
+        except SchemaError:
```

Comment:
This exception is in the right place IMHO, as the schema is a valid mapping, but when we try to validate a document against it, it fails because the schema is not valid.

Reply:
Ahhhh, I get it. That's really actually great. We should be checking the schema first, if invalid then no reason going further. Gotcha :100: